st2actions/st2actions/runners/remote_script_runner.py

```diff
@@ -78,3 +84,116 @@ def _get_remote_action(self, action_parameters):
+
+    def _run(self, remote_action):
+        try:
+            copy_results = self._copy_artifacts(remote_action)
+        except:
+            # If for whatever reason there is a top level exception,
+            # we just bail here.
+            error = 'Failed copying content to remote boxes.'
+            LOG.exception(error)
+            _, ex, tb = sys.exc_info()
+            copy_results = self._generate_error_results(' '.join([error, str(ex)]), tb)
+            return copy_results
+        else:
+            try:
+                exec_results = self._run_script_on_remote_host(remote_action)
+                try:
+                    remote_dir = remote_action.get_remote_base_dir()
+                    LOG.info('Deleting dir %s', remote_dir)
+                    delete_results = self._parallel_ssh_client.delete_dir(path=remote_dir,
+                                                                          force=True)
+                    LOG.info('Delete results: %s', delete_results)
+                except:
+                    LOG.exception('Failed deleting remote dir: %s', remote_dir)
+                finally:
+                    return exec_results
+            except:
+                error = 'Failed executing script on remote boxes.'
+                LOG.exception(error)
+                _, ex, tb = sys.exc_info()
+                exec_results = self._generate_error_results(' '.join([error, str(ex)]), tb)
+                return exec_results
+
+    def _copy_artifacts(self, remote_action):
+        # First create remote execution directory.
+        LOG.info('mkdir: %s', remote_action.get_remote_base_dir())
+        self._parallel_ssh_client.mkdir(path=remote_action.get_remote_base_dir())
+        # Copy the script to remote dir in remote host.
+        LOG.info('Copy: %s to %s', remote_action.get_local_script_abs_path(),
+                 remote_action.get_remote_script_abs_path())
+        self._parallel_ssh_client.put(local_path=remote_action.get_local_script_abs_path(),
+                                      remote_path=remote_action.get_remote_script_abs_path(),
+                                      mirror_local_mode=True)
+        # If `lib` exist for the script, copy that to remote host.
+        if os.path.exists(remote_action.get_local_libs_path_abs()):
+            LOG.info('Copy: %s to %s', remote_action.get_local_libs_path_abs(),
+                     remote_action.get_remote_libs_path_abs())
+            self._parallel_ssh_client.put(local_path=remote_action.get_local_libs_path_abs(),
+                                          remote_path=remote_action.get_remote_base_dir(),
+                                          mirror_local_mode=True)
+
+    def _run_script_on_remote_host(self, remote_action):
+        command = remote_action.get_full_command_string()
+        LOG.info('Command to run: %s', command)
+        results = self._parallel_ssh_client.run(command, timeout=remote_action.get_timeout(),
```

Comment:
Do we make sure the script on the remote host is executable?

I made this change a while ago in the fabric runner since the local file wasn't executable, it would blow it when we tried to execute it on the remote host.